commit 001 Домашняя работа к семинару 1

medvedit opened the pull request.

medvedit committed:
commit 001 Домашняя работа к семинару 1

medvedit committed:
commit 002 Добавил переопределенный метод equals(), но без использования проверки id.

medvedit committed:
commit 003 Добавил методы, изменил написанные ранее после просмотра семинара №2.

medvedit committed:
commit 004 Домашняя работа к семинару 2

medvedit committed:
commit 005 Изменил домашнюю работу к семинару 2. Добавление try-with-resources, перенос методов saveFile и readFile в отдельный класс, добавление интерфейса, Путь для сохранения прописан в интерфейсе.

Liberate520 commented on src/ru/medved_sa/tree_family/save/Writable.java:
```diff
@@ -0,0 +1,12 @@
+import java.io.Serializable;
+
+public interface Writable {
+    String filePath = "src/ru/medved_sa/tree_family/save/familyTree.out";
```
Лучше переменную хранить в классах, а не в интерфейсах. Потому что не для всех эта переменная может быть актуальна. Например если класс будет записывать данные в БД

medvedit replied:
Исправил. Буду благодарен за комментарий.

medvedit committed:
commit 005 Изменил домашнюю работу к семинару 2 после комментария преподавателя. Создал отдельный класс, где храниться адрес для backup данных.

medvedit committed:
commit 006 Изменил способ передачи адреса для сохранения файла.

medvedit commented:
Можно проверить правильность решения в commit 006 ?

medvedit committed:
commit 007 Домашняя работа к семинару 3.

medvedit committed:
commit 008 Промежуточное сохранение - удаление лишних комментариев, поименовал имена некоторых переменных. Изменений в коде - нет!

medvedit committed:
commit 009 Реализация обобщений

medvedit committed:
commit 010 Реализация MVP

Liberate520 commented on src/ru/medved_sa/tree_family/model/FamilyTreeService.java:
```diff
@@ -0,0 +1,59 @@
+        return "Человек успешно добавлен в дерево";
+    }
+
+    public FamilyTreeService(Writable writable) {
+        this.writable = writable;
+    }
```
У вас инициализация поля writable выполняется только в этом месте, Однако этот конструктор вы не используете. Следовательно сохранение и загрузка из консоли у вас не работает

Liberate520 commented on src/ru/medved_sa/tree_family/presenter/Presenter.java:
```diff
@@ -0,0 +1,44 @@
+        service = new FamilyTreeService();
+    }
+
+    public void addHuman(String name, String gender, String birthDate) {
+        String answer = service.addHuman(name, gender, birthDate);
+        view.print(answer);
+    }
```
Нет возможности созданий родственных связей

medvedit committed:
commit 011 SOLID

Реализация чтения семейного дерева из файла familyTree.out
Реализация сохранения изменений в семейном дереве
Добавление/реализация создания связей
Рефакторинг проекта с семейным деревом с учетом принципов SOLID

medvedit committed:
commit 012 реализация сортировки по ID

* Основные изменения коснулись возможности отсортировать список
семейного дерева по ID человека. Точнее убрал метод NoSortedList
заменив его на метод sortById протянув его от model/human/comparator
через Service.java, Presenter.java во view
* Не значительные изменения в выводе текста в консоль
* Изменение очередности вывода доступных запросов для пользователя в консоли.

Liberate520 commented on src/ru/medved_sa/tree_family/presenter/Presenter.java:
```diff
@@ -0,0 +1,59 @@
+    }
+
+    public boolean setParent(int personId, int parentId){
+        if (service.checkId(personId) & service.checkId(parentId)) {
+            service.setParent(personId, parentId);
```
Работа сервиса, а не презентера

Liberate520 replied:
И используйте двойной амперсант для операторов И

Liberate520 commented on src/ru/medved_sa/tree_family/model/Service.java:
```diff
@@ -0,0 +1,69 @@
+    }
+
+    public boolean readFile(String filePath) {
+        FamilyTree<Human> tempTree = new FamilyTree<>();
```
лишнее действие

Liberate520 commented on src/ru/medved_sa/tree_family/model/Service.java:
```diff
@@ -0,0 +1,69 @@
+    public boolean readFile(String filePath) {
+        FamilyTree<Human> tempTree = new FamilyTree<>();
+        tempTree = writable.readFile(filePath);
+        if (writable == null) {
```
если writable null, то приложение упадет еще строчкой выше, где вы вызываете у writable метод чтения

medvedit committed:
commit 013 исправление недочетов указанных преподавателем.

* Presenter.java в методе setParent().
* Service.java в методе readFile().

* В классе Main - не влияющие на работу приложения изменения.
  Закомментировал не используемые методы.
* Файл familyTree.out содержит сохранения данных о ранее
  созданных сущностях семейного дерева.

medvedit committed:
commit 014 - Урок 7. Итоговый проект.

* Спасибо вам большое! Вы очень хороший преподаватель!

Liberate520 commented on src/ru/medved_sa/tree_family/model/service/Service.java:
```diff
@@ -0,0 +1,75 @@
+    }
+
+    public boolean setParent(int person_id, int parent_id) {
+        Human human = activeTree.getById(person_id);
```
наверное строчка должна быть где то внутри if. Иначе как то странно. Сначала используем, а потом проверяем

medvedit committed:
commit 015 - Исправление замечания от преподавателя.
